### DISCUSSION TITLE
How can I get the cursor position?

### DISCUSSION THREAD

**Discussion by saruman9** [Vector35/binaryninja-api#3009] [API] [answered]
I have the selected (or highlighted?) token (type, variable, constant, etc) under the cursor after clicking. How can I get a type or a position or a name of the token?

For example:

![image](https://user-images.githubusercontent.com/4244396/157836691-d60d1186-6fc8-4714-b579-8c4cc14a635c.png)
> Name: `ecx`, position: `address`, `offset`?, type: `register` or `variable` for HLIL

![image](https://user-images.githubusercontent.com/4244396/157836699-b733116d-5e6d-4ebe-87b0-bf4c5fa7a688.png)
> Name: `var_40_1`, position: `address`, `offset`?, type: `variable` for HLIL

![image](https://user-images.githubusercontent.com/4244396/157836708-52bca485-ba53-4ff0-9563-632894683d9e.png)
> Name: `std::__cxx11::string`, position: `address`, `offset`?, type: `type`

**Comment by psifertex** [comment, marked as answer]
You'll want to use a UI plugin for this. You'll receive a UIContext object that will contain information like the view, selection, etc. snippets is one example plugin that does this but there are others. 

https://github.com/Vector35/snippets/blob/master/__init__.py
(See specifically, the uses of the UIAction module) 

note that this module isn't documented the normal way because it's exposed via PySide. 

https://api.binary.ninja/cpp/class_u_i_context.html

In short, look for example plugins using the registerNotification API
